libwasmvm/src/api.rs

```diff
@@ -254,8 +262,46 @@ impl BackendApi for GoApi {
             Err(e) => return (Err(BackendError::unknown(e.to_string())), gas_info),
         };
         callee_instance.env.set_serialized_env(&contract_env);
-        callee_instance.set_storage_readonly(caller_env.is_storage_readonly());
         gas_info.cost += instantiate_cost;
+        // set read-write permission to callee instance
+        let callee_info = FunctionMetadata {
+            module_name: String::from(&func_info.module_name),
+            name: "_callee_func_list".to_string(),
```

Comment:
It is a function's name, so starting with "_list" is better. And, would you use a clearer name like "callable_points" rather than "callee_func"? This is because the function is defined in callee contracts, and the listed functions are not "callee_function" from the point of them.